app/stac_api/models.py

```diff
@@ -106,12 +106,23 @@ def __str__(self):
 
 
 class Provider(models.Model):
+    collection = models.ForeignKey(
+        'stac_api.Collection',
+        on_delete=models.CASCADE,
+        related_name='providers',
+        related_query_name='provider'
+    )
     name = models.CharField(blank=False, max_length=200)
     description = models.TextField(blank=True, null=True)
-    roles = ArrayField(models.CharField(max_length=9))
-    # possible roles are licensor, producer, processor or host. Probably it might sense
+    # possible roles are licensor, producer, processor or host
+    roles = ArrayField(models.CharField(max_length=9), help_text=_(
+        "Comma-separated list of roles. Possible values are 'licensor', "
```

Comment:
Probably it would make sense to move the `allowed_roles` from the Provider's clean()-method up here and do something like:
```python
`    allowed_roles = ['licensor', 'producer', 'processor', 'host']
    # possible roles are licensor, producer, processor or host
    roles = ArrayField(models.CharField(max_length=9), help_text=_(
        "Comma-separated list of roles. Possible values are: " +
        ', '.join(allowed_roles)))
```

Just in case the definition of `allowed_roles` should change, it would be automatically updated in the help text.

(in the clean() a `self` would need to be added before the `allowed_roles`:
`if role not in self.allowed_roles:`)

Reply:
:+1: I changed this